Cineaste/Models/StoredMovie.swift

```diff
@@ -20,9 +32,129 @@ class StoredMovie: NSManagedObject {
         if let moviePoster = movie.poster {
             poster = UIImageJPEGRepresentation(moviePoster, 1)
         }
-        voteAverage = movie.voteAverage
+        voteAverage = movie.voteAverage as NSDecimalNumber
+        voteCount = movie.voteCount
         runtime = movie.runtime
         releaseDate = movie.releaseDate
         watched = false
+        watchedDate = nil
+        listPosition = 0
+    }
+
+    convenience init(context moc: NSManagedObjectContext) {
+        let name = String(describing: type(of: self))
+        guard let entity = NSEntityDescription.entity(forEntityName: name, in: moc) else {
+            fatalError("Unable to create entity description with \(name)")
+        }
+
+        self.init(entity: entity, insertInto: moc)
+    }
+
+    enum CodingKeys: String, CodingKey {
```

Comment:
I would prefer them being in the same order as they are initialised in the constructors.

Reply:
Fixed that!